src/components/elements/Map-mapbox/utils.ts

```diff
@@ -897,6 +898,48 @@ export const formatCommentaryDate = (date: Date | null | undefined): string => {
     : "Unknown";
 };
 
+// New utility functions for SitePolygonFullDto
+export function parsePolygonDataV3(sitePolygonData: SitePolygonFullDto[] | undefined) {
+  return (sitePolygonData ?? []).reduce((acc: Record<string, string[]>, data: SitePolygonFullDto) => {
+    if (data.status && data.polygonUuid !== null) {
+      if (!acc[data.status]) {
```

Comment:
Please use `==` for null comparisons (note I change `!==` to `!=` as well, to catch `undefined`)
```
if (data.status != null && data.polygonUuid != null) {
  if (acc[data.status] == null) {
```